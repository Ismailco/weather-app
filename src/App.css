@import url('https://fonts.googleapis.com/css2?family=Fira+Sans:ital,wght@0,300;0,400;0,700;1,400;1,700&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  font-family: 'Fira Sans', Arial, Helvetica, sans-serif;
  background-color: #eb4c89;
  min-height: 100vh;
}

.header {
  font-size: 12px;
}
