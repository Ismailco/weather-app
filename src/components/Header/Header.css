.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 30px;
}

.current-tab {
  font-size: 20px;
  color: #ffe7ee;
}

.header-btn {
  background-color: transparent;
  border: none;
  outline: none;
  cursor: pointer;
}

.fa-icon {
  font-size: 25px;
  color: #ffe7ee;
  background-color: transparent;
  margin: 5px 10px;
}

.nav {
  display: flex;
  justify-content: flex-end;
  width: 1px;
}

.search-input {
  color: #ffe7ee;
  padding-left: 10px;
  font-weight: 700;
  font-size: 15px;
  background-image: url(../../images/search.svg);
  background-repeat: no-repeat;
  background-position: right;
  background-size: 25px;
  opacity: 1;
  background-color: transparent;
  width: 30px;
  border: 2px solid transparent;
  outline: none;
  border-radius: 100px;
  cursor: pointer;
  transition: width 200ms ease-in-out;
}

.search-input:valid,
.search-input:focus {
  border: 2px solid #ffe7ee;
  width: 200px;
  margin-left: -170px;
  cursor: text;
  background-image: none;
}
