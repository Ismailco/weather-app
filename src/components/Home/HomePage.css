.city-unit {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    flex-basis: 30vw;
}