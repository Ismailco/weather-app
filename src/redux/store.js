import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import reducer from './city/city';

const root = combineReducers({
  citys: reducer,
});
const store = createStore(root, applyMiddleware(logger, thunk));
store.dispatch(reducer);
export default store;
