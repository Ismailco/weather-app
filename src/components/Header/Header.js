import React from 'react';
import './Header.css';
import { useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft, faGear, faHome } from '@fortawesome/free-solid-svg-icons';
import { useNavigate } from 'react-router-dom';
import { removeData, dispatchGetData } from '../../redux/city/city';

function Header() {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const onKeyPress = (e) => {
    if (e.key === 'Enter') {
      dispatch(removeData());
      dispatch(dispatchGetData(e.target.value));
      e.target.value = '';
    }
  };

  return (
    <div className="header">
      <div className="home">
        <button type="button" className="header-btn" onClick={() => navigate(-1)}>
          <FontAwesomeIcon className="fa-icon" icon={faArrowLeft} />
        </button>
        <a href="/">
          <FontAwesomeIcon className="fa-icon" icon={faHome} />
        </a>
      </div>
      <p className="current-tab">Global Forecast</p>
      <div className="nav">
        <input className="search-input" type="text" required onKeyPress={(e) => onKeyPress(e)} />
        <a href="/">
          <FontAwesomeIcon className="fa-icon" icon={faGear} />
        </a>
      </div>
    </div>
  );
}

export default Header;
