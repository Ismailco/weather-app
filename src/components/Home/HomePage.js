import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { dispatchGetData } from '../../redux/city/city';
import City from '../City/City';
import './HomePage.css';

function HomePage() {
  const city = useSelector((state) => state.city);
  const dispatch = useDispatch();

  useEffect(() => {
    if (city.length === 0) {
      dispatch(dispatchGetData('ma'));
    }
  }, []);

  return (
    <div>
      {city.length === 0 && <h2 className="empty-list">The is no Weather Data for now!</h2>}
      {city.length > 0 && (
        <div>
          {/* <h2>Weather in: {}</h2> */}
          <section className="city-unit">
            {city.map((city) => (
              <City city={city} weather={city.consolidated_weather} key={city.woeid} />
            ))}
          </section>
        </div>
      )}
    </div>
  );
}

export default HomePage;
