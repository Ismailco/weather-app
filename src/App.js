import './App.css';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Footer from './components/Footer/Footer';
import Header from './components/Header/Header';
import store from './redux/store';
import HomePage from './components/Home/HomePage';
import CityPage from './components/CityPage/CityPage';

function App() {
  return (
    <BrowserRouter className="App">
      <Provider store={store}>
        <Header />
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/city/:woeid/:title" element={<CityPage />} />
        </Routes>
        <Footer />
      </Provider>
    </BrowserRouter>
  );
}

export default App;
