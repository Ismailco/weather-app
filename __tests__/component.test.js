/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../src/redux/city/city';
import HomePage from '../src/components/Home/HomePage';
import Header from '../src/components/Header/Header';
import App from '../src/App';
import 'isomorphic-fetch';

const data = {
  title: 'Casablanca',
  location_type: 'City',
  woeid: 1532755,
  latt_long: '33.596611,-7.619340',
  timezone: 'Africa/Casablanca',
  consolidated_weather: [
    {
      id: 5950947143450624,
      weather_state_name: 'Showers',
      weather_state_abbr: 's',
      wind_direction_compass: 'WSW',
      created: '2022-04-22T13:46:58.741177Z',
      applicable_date: '2022-04-22',
      min_temp: 11.475000000000001,
      max_temp: 20.615000000000002,
      the_temp: 19.455,
      wind_speed: 14.286615712818474,
      wind_direction: 246.17416337531438,
      air_pressure: 1010.0,
      humidity: 62,
      visibility: 12.152777777777779,
      predictability: 73,
    },
    {
      id: 5593192406712320,
      weather_state_name: 'Light Rain',
      weather_state_abbr: 'lr',
      wind_direction_compass: 'W',
      created: '2022-04-22T13:47:02.029156Z',
      applicable_date: '2022-04-23',
      min_temp: 14.48,
      max_temp: 19.445,
      the_temp: 18.855,
      wind_speed: 9.501171875818551,
      wind_direction: 262.8672855344178,
      air_pressure: 1014.5,
      humidity: 60,
      visibility: 12.474026684164478,
      predictability: 75,
    },
    {
      id: 5520167023935488,
      weather_state_name: 'Heavy Cloud',
      weather_state_abbr: 'hc',
      wind_direction_compass: 'N',
      created: '2022-04-22T13:47:04.742683Z',
      applicable_date: '2022-04-24',
      min_temp: 12.715,
      max_temp: 19.465,
      the_temp: 18.91,
      wind_speed: 6.416891456409236,
      wind_direction: 11.0,
      air_pressure: 1015.0,
      humidity: 57,
      visibility: 12.683118090352341,
      predictability: 71,
    },
    {
      id: 6056626055282688,
      weather_state_name: 'Light Cloud',
      weather_state_abbr: 'lc',
      wind_direction_compass: 'N',
      created: '2022-04-22T13:47:08.056645Z',
      applicable_date: '2022-04-25',
      min_temp: 13.11,
      max_temp: 19.759999999999998,
      the_temp: 19.09,
      wind_speed: 6.201440273599512,
      wind_direction: 1.941307521419306,
      air_pressure: 1013.5,
      humidity: 65,
      visibility: 12.397287341923168,
      predictability: 70,
    },
    {
      id: 6719092313554944,
      weather_state_name: 'Light Cloud',
      weather_state_abbr: 'lc',
      wind_direction_compass: 'N',
      created: '2022-04-22T13:47:11.352931Z',
      applicable_date: '2022-04-26',
      min_temp: 13.2,
      max_temp: 20.125,
      the_temp: 19.695,
      wind_speed: 5.956837610506642,
      wind_direction: 357.56020527692584,
      air_pressure: 1015.5,
      humidity: 61,
      visibility: 12.817644953471724,
      predictability: 70,
    },
    {
      id: 4615331768172544,
      weather_state_name: 'Heavy Cloud',
      weather_state_abbr: 'hc',
      wind_direction_compass: 'N',
      created: '2022-04-22T13:47:14.058505Z',
      applicable_date: '2022-04-27',
      min_temp: 14.2,
      max_temp: 20.425,
      the_temp: 19.82,
      wind_speed: 5.626198202497415,
      wind_direction: 4.999999999999999,
      air_pressure: 1020.0,
      humidity: 74,
      visibility: 9.999726596675416,
      predictability: 71,
    },
  ],
  time: '2022-04-22T14:48:13.528006Z',
  sun_rise: '2022-04-22T05:51:52.202109Z',
  sun_set: '2022-04-22T19:06:38.314590Z',
  timezone_name: 'LMT',
  parent: {
    title: 'Morocco',
    location_type: 'Country',
    woeid: 23424893,
    latt_long: '31.434200,-6.402450',
  },
  sources: [
    {
      title: 'BBC',
      slug: 'bbc',
      url: 'http://www.bbc.co.uk/weather/',
      crawl_rate: 360,
    },
    {
      title: 'Forecast.io',
      slug: 'forecast-io',
      url: 'http://forecast.io/',
      crawl_rate: 480,
    },
    {
      title: 'Met Office',
      slug: 'met-office',
      url: 'http://www.metoffice.gov.uk/',
      crawl_rate: 180,
    },
    {
      title: 'OpenWeatherMap',
      slug: 'openweathermap',
      url: 'http://openweathermap.org/',
      crawl_rate: 360,
    },
    {
      title: 'Weather Underground',
      slug: 'wunderground',
      url: 'https://www.wunderground.com/?apiref=fc30dc3cd224e19b',
      crawl_rate: 720,
    },
    {
      title: 'World Weather Online',
      slug: 'world-weather-online',
      url: 'http://www.worldweatheronline.com/',
      crawl_rate: 360,
    },
  ],
};

// I created the same actions and redeucer here instead of importing them
// Action
const GET_DATA = 'weatherApp/city/GET_DATA';
export const getData = (citys) => ({
  type: GET_DATA,
  payload: {
    citys,
  },
});

// Create store
const rootReducer = combineReducers({
  city: reducer,
});
const store = createStore(rootReducer, applyMiddleware(thunk));

afterEach(cleanup);
store.dispatch(getData(data));

describe('Countries List ', () => {
  it('renders CityPage correctly', () => {
    const countries = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(countries).toMatchSnapshot();
  });
  it('renders HomePage correctly', () => {
    const regions = renderer
      .create(
        <Provider store={store}>
          <Router>
            <HomePage />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(regions).toMatchSnapshot();
  });
  test('renders App correctly', () => {
    const pollutions = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(pollutions).toMatchSnapshot();
  });
});
