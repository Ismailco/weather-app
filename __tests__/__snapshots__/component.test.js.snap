// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries List  renders App correctly 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="home"
    >
      <button
        className="header-btn"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-arrow-left fa-icon"
          data-icon="arrow-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M447.1 256C447.1 273.7 433.7 288 416 288H109.3l105.4 105.4c12.5 12.5 12.5 32.75 0 45.25C208.4 444.9 200.2 448 192 448s-16.38-3.125-22.62-9.375l-160-160c-12.5-12.5-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L109.3 224H416C433.7 224 447.1 238.3 447.1 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <a
        href="/"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-house fa-icon"
          data-icon="house"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </div>
    <p
      className="current-tab"
    >
      Global Forecast
    </p>
    <div
      className="nav"
    >
      <input
        className="search-input"
        onKeyPress={[Function]}
        required={true}
        type="text"
      />
      <button
        className="header-btn"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-gear fa-icon"
          data-icon="gear"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M495.9 166.6C499.2 175.2 496.4 184.9 489.6 191.2L446.3 230.6C447.4 238.9 448 247.4 448 256C448 264.6 447.4 273.1 446.3 281.4L489.6 320.8C496.4 327.1 499.2 336.8 495.9 345.4C491.5 357.3 486.2 368.8 480.2 379.7L475.5 387.8C468.9 398.8 461.5 409.2 453.4 419.1C447.4 426.2 437.7 428.7 428.9 425.9L373.2 408.1C359.8 418.4 344.1 427 329.2 433.6L316.7 490.7C314.7 499.7 307.7 506.1 298.5 508.5C284.7 510.8 270.5 512 255.1 512C241.5 512 227.3 510.8 213.5 508.5C204.3 506.1 197.3 499.7 195.3 490.7L182.8 433.6C167 427 152.2 418.4 138.8 408.1L83.14 425.9C74.3 428.7 64.55 426.2 58.63 419.1C50.52 409.2 43.12 398.8 36.52 387.8L31.84 379.7C25.77 368.8 20.49 357.3 16.06 345.4C12.82 336.8 15.55 327.1 22.41 320.8L65.67 281.4C64.57 273.1 64 264.6 64 256C64 247.4 64.57 238.9 65.67 230.6L22.41 191.2C15.55 184.9 12.82 175.3 16.06 166.6C20.49 154.7 25.78 143.2 31.84 132.3L36.51 124.2C43.12 113.2 50.52 102.8 58.63 92.95C64.55 85.8 74.3 83.32 83.14 86.14L138.8 103.9C152.2 93.56 167 84.96 182.8 78.43L195.3 21.33C197.3 12.25 204.3 5.04 213.5 3.51C227.3 1.201 241.5 0 256 0C270.5 0 284.7 1.201 298.5 3.51C307.7 5.04 314.7 12.25 316.7 21.33L329.2 78.43C344.1 84.96 359.8 93.56 373.2 103.9L428.9 86.14C437.7 83.32 447.4 85.8 453.4 92.95C461.5 102.8 468.9 113.2 475.5 124.2L480.2 132.3C486.2 143.2 491.5 154.7 495.9 166.6V166.6zM256 336C300.2 336 336 300.2 336 255.1C336 211.8 300.2 175.1 256 175.1C211.8 175.1 176 211.8 176 255.1C176 300.2 211.8 336 256 336z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>,
  <div>
    <div
      className="loading-wraper"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-spinner loading"
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48C304 74.51 282.5 96 256 96C229.5 96 208 74.51 208 48C208 21.49 229.5 0 256 0C282.5 0 304 21.49 304 48zM304 464C304 490.5 282.5 512 256 512C229.5 512 208 490.5 208 464C208 437.5 229.5 416 256 416C282.5 416 304 437.5 304 464zM0 256C0 229.5 21.49 208 48 208C74.51 208 96 229.5 96 256C96 282.5 74.51 304 48 304C21.49 304 0 282.5 0 256zM512 256C512 282.5 490.5 304 464 304C437.5 304 416 282.5 416 256C416 229.5 437.5 208 464 208C490.5 208 512 229.5 512 256zM74.98 437C56.23 418.3 56.23 387.9 74.98 369.1C93.73 350.4 124.1 350.4 142.9 369.1C161.6 387.9 161.6 418.3 142.9 437C124.1 455.8 93.73 455.8 74.98 437V437zM142.9 142.9C124.1 161.6 93.73 161.6 74.98 142.9C56.24 124.1 56.24 93.73 74.98 74.98C93.73 56.23 124.1 56.23 142.9 74.98C161.6 93.73 161.6 124.1 142.9 142.9zM369.1 369.1C387.9 350.4 418.3 350.4 437 369.1C455.8 387.9 455.8 418.3 437 437C418.3 455.8 387.9 455.8 369.1 437C350.4 418.3 350.4 387.9 369.1 369.1V369.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>,
  <div
    className="footer"
  >
    <p
      className="footer-content"
    >
      Created with
      <span
        className="hearts"
      >
        ♥
      </span>
      and
      <span
        className="coffee"
      >
        ☕
      </span>
      in Morocco,
      <br />
      This App under GPL-3 License.
    </p>
  </div>,
]
`;

exports[`Countries List  renders CityPage correctly 1`] = `
<div
  className="header"
>
  <div
    className="home"
  >
    <button
      className="header-btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-icon"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M447.1 256C447.1 273.7 433.7 288 416 288H109.3l105.4 105.4c12.5 12.5 12.5 32.75 0 45.25C208.4 444.9 200.2 448 192 448s-16.38-3.125-22.62-9.375l-160-160c-12.5-12.5-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L109.3 224H416C433.7 224 447.1 238.3 447.1 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <a
      href="/"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house fa-icon"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <p
    className="current-tab"
  >
    Global Forecast
  </p>
  <div
    className="nav"
  >
    <input
      className="search-input"
      onKeyPress={[Function]}
      required={true}
      type="text"
    />
    <button
      className="header-btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-gear fa-icon"
        data-icon="gear"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M495.9 166.6C499.2 175.2 496.4 184.9 489.6 191.2L446.3 230.6C447.4 238.9 448 247.4 448 256C448 264.6 447.4 273.1 446.3 281.4L489.6 320.8C496.4 327.1 499.2 336.8 495.9 345.4C491.5 357.3 486.2 368.8 480.2 379.7L475.5 387.8C468.9 398.8 461.5 409.2 453.4 419.1C447.4 426.2 437.7 428.7 428.9 425.9L373.2 408.1C359.8 418.4 344.1 427 329.2 433.6L316.7 490.7C314.7 499.7 307.7 506.1 298.5 508.5C284.7 510.8 270.5 512 255.1 512C241.5 512 227.3 510.8 213.5 508.5C204.3 506.1 197.3 499.7 195.3 490.7L182.8 433.6C167 427 152.2 418.4 138.8 408.1L83.14 425.9C74.3 428.7 64.55 426.2 58.63 419.1C50.52 409.2 43.12 398.8 36.52 387.8L31.84 379.7C25.77 368.8 20.49 357.3 16.06 345.4C12.82 336.8 15.55 327.1 22.41 320.8L65.67 281.4C64.57 273.1 64 264.6 64 256C64 247.4 64.57 238.9 65.67 230.6L22.41 191.2C15.55 184.9 12.82 175.3 16.06 166.6C20.49 154.7 25.78 143.2 31.84 132.3L36.51 124.2C43.12 113.2 50.52 102.8 58.63 92.95C64.55 85.8 74.3 83.32 83.14 86.14L138.8 103.9C152.2 93.56 167 84.96 182.8 78.43L195.3 21.33C197.3 12.25 204.3 5.04 213.5 3.51C227.3 1.201 241.5 0 256 0C270.5 0 284.7 1.201 298.5 3.51C307.7 5.04 314.7 12.25 316.7 21.33L329.2 78.43C344.1 84.96 359.8 93.56 373.2 103.9L428.9 86.14C437.7 83.32 447.4 85.8 453.4 92.95C461.5 102.8 468.9 113.2 475.5 124.2L480.2 132.3C486.2 143.2 491.5 154.7 495.9 166.6V166.6zM256 336C300.2 336 336 300.2 336 255.1C336 211.8 300.2 175.1 256 175.1C211.8 175.1 176 211.8 176 255.1C176 300.2 211.8 336 256 336z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Countries List  renders HomePage correctly 1`] = `
<div>
  <div>
    <section
      className="city-unit"
    >
      <a
        className="city"
        href="/city/1532755/Casablanca"
        onClick={[Function]}
      >
        <div
          className="city-weather-info"
        >
          <img
            alt="Casablanca"
            className="city-state"
            src="https://www.metaweather.com/static/img/weather/s.svg"
          />
          <div
            className="city-state-info"
          >
            <p
              className="city-the-temp"
            >
              19
              °
            </p>
            <p
              className="sep"
            />
            <p>
              Showers
            </p>
          </div>
        </div>
        <div
          className="city-about-info"
        >
          <h2
            className="city-name"
          >
            Casablanca
          </h2>
          <p
            className="city-type"
          >
            City, Morocco
          </p>
          <p
            className="gps-cord"
          >
            GPS:
            33.596611,-7.619340
          </p>
        </div>
      </a>
    </section>
  </div>
</div>
`;
